backends/dpdk/dpdkProgram.cpp

```diff
@@ -286,12 +286,15 @@ void ConvertToDpdkParser::getCondVars(const IR::Expression *sv, const IR::Expres
                 "%1%, Select expression wider than 64-bit is not permitted", sv);
         return;
     }
+    auto width = sv->type->width_bits();
+    unsigned byteAlignedWidth = width + 8 - width % 8;
```

Comment:
(width + 7) & 0x7  is the correct method
what you have done is not correct

Reply:
It should be (width + 7) & (-7).
or
((width + 7) >> 3) << 3

Reply:
> (-7).

you meant ~7

Reply:
yes, bitwise not to reset least three bits.

Reply:
yes, thanks for catching it, it wasn't handling already aligned values.